app/db/quiz_db.py

```
@@ -0,0 +1,126 @@
+from datetime import datetime
+from bson import ObjectId
+from pymongo import MongoClient
+
+
+class QuizDB:
+    """
+    Class to handle all the database operations related to quizzes (currently stored on Mongo Atlas)
+    """
+
+    def __init__(self, db: MongoClient) -> None:
+        self.__db = db
+
+    def __generate_objectid_for_time(self, time):
+        """
+        Generates a Mongo object ID for a given time
+        """
+        timeId = str(ObjectId.from_datetime(time))
+        return timeId
+
+    def get_live_quiz_stats(self, quiz_id, start_date=None, end_date=None):
+        """
+        Returns daywise report for a given quiz ID by using an aggregation pipeline
+
+        params:
+            quiz_id: The quiz ID
+            start_date (optional): The start date for the report
+            end_date (optional): The end date for the report
+        """
+
+        # Get quiz details
+        quiz = self.__db.quiz.quizzes.find_one({"_id": quiz_id})
```

if quiz is none, then quiz_id is wrong --> we can throw a meaningful error. currently getting internal server error for random quiz id https://reports-staging.avantifellows.org/reports/live_quiz_report/657812838be7909220ffdbdex